frontend/src/libs/googleAnalytics/hooks/useTrackPageTimeGA.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useRef } from "react";
+import ReactGA from "react-ga4";
+import { useLocation } from "react-router";
+
+const normalizePath = (pathname: string) => {
+  if (/^\/detail\/\d+/.test(pathname)) {
+    return "/detail/[id]";
+  }
+
+  return pathname;
+};
+
+const useTrackPageTimeGA = () => {
+  const location = useLocation();
+  const prevPathRef = useRef(location.pathname);
+  const startTimeRef = useRef(Date.now());
+  const isFirstRunRef = useRef(true);
+
+  useEffect(() => {
+    if (isFirstRunRef.current) {
+      isFirstRunRef.current = false;
+      return;
+    }
+
+    const now = Date.now();
+    const stayTime = now - startTimeRef.current;
+
+    if (prevPathRef.current !== location.pathname) {
+      ReactGA.event({
+        category: "User Engagement",
+        action: "time_stay_per_page",
```

Comment:
```suggestion
        category: "User Engagement",
        action: "time_stay_per_page",
```
이런 값들은 상수화로 관리하면 휴면 에러도 방지하고 한 곳에 응집되어 유지보수 하기도 좋을 것 같다고 생각하는데 세오는 어떻게 생각하시나요??🤔